nova/virt/vmwareapi/vmops.py

```diff
@@ -1535,6 +1516,30 @@ def _destroy_instance(self, context, instance, destroy_disks=True):
                 self._session._call_method(self._session.vim,
                                            "UnregisterVM", vm_ref)
                 LOG.debug("Unregistered the VM", instance=instance)
+
+                if server_group_info:
+                    with lockutils.lock('vmware-vm-group-policy'):
```

Comment:
You need to set the same `lock_file_prefix` as `cluster_util.update_placement()` uses. Please look at `nova.utils` for the `@synchronized` decorator.